### ISSUE TITLE
[Boost-regex] Build error "stdio.h:83:8: error: 'FILE' does not name a type"

### ISSUE THREAD

**Issue by mbartu-r7** [microsoft/vcpkg#41037]
### Operating system

MacOS Sonoma 14.6.1 - Intel-based Mac only

### Compiler

gcc 12.4.0 (Homebrew)

### Steps to reproduce the behavior

```Shell
cmake .
```

### Failure logs

Installing 40/64 boost-regex:x64-osx@1.85.0#1...
Building boost-regex:x64-osx@1.85.0#1...
/Users/alpha/vcpkg/buildtrees/versioning_/versions/boost-regex/6f5d77fa3adaafae00b56ca0072d56ef4c4836a4: info: installing overlay port from here
-- Downloading https://github.com/boostorg/regex/archive/boost-1.85.0.tar.gz -> boostorg-regex-boost-1.85.0.tar.gz...
-- Extracting source /Users/alpha/vcpkg/downloads/boostorg-regex-boost-1.85.0.tar.gz
-- Applying patch compat.diff
-- Using source at /Users/alpha/vcpkg/buildtrees/boost-regex/src/ost-1.85.0-afd9b213e1.clean
-- Configuring x64-osx
-- Building x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/local/Cellar/cmake/3.30.3/bin/cmake --build . --config Debug --target install -- -v -j3
    Working Directory: /Users/alpha/vcpkg/buildtrees/boost-regex/x64-osx-dbg
    See logs for more information:
      /Users/alpha/vcpkg/buildtrees/boost-regex/install-x64-osx-dbg-out.log

Call Stack (most recent call first):
  /Users/alpha/code/UnmanagedInfra/.m/vcpkg_installed/x64-osx/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  /Users/alpha/code/UnmanagedInfra/.m/vcpkg_installed/x64-osx/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  /Users/alpha/code/UnmanagedInfra/.m/vcpkg_installed/x64-osx/share/vcpkg-boost/boost-install.cmake:65 (vcpkg_cmake_install)
  buildtrees/versioning_/versions/boost-regex/6f5d77fa3adaafae00b56ca0072d56ef4c4836a4/portfile.cmake:15 (boost_configure_and_install)
  scripts/ports.cmake:192 (include)
  
  
[install-x64-osx-dbg-out.log](https://github.com/user-attachments/files/17032022/install-x64-osx-dbg-out.log)

### Additional context

Same project & compiler configurations works on Arm-based Mac

**Comment by dg0yt** [comment]
~~~
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/Availability.h:178,
                 from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/_stdlib.h:61,
                 from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/stdlib.h:58,
                 from /usr/local/Cellar/gcc@12/12.4.0/include/c++/12/cstdlib:75,
                 from /usr/local/Cellar/gcc@12/12.4.0/include/c++/12/stdlib.h:36,
                 from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/assert.h:44:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/_wchar.h:215:47: error: attributes after parenthesized initializer ignored [-fpermissive]
  215 | wchar_t *fgetwln(FILE * __restrict, size_t *) __OSX_AVAILABLE_STARTING(__MAC_10_7, __IPHONE_4_3);
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/_wchar.h:215:47: error: expression list treated as compound expression in initializer [-fpermissive]
In file included from /usr/local/Cellar/gcc@12/12.4.0/include/c++/12/ext/string_conversions.h:43,
                 from /usr/local/Cellar/gcc@12/12.4.0/include/c++/12/bits/basic_string.h:3972,
                 from /usr/local/Cellar/gcc@12/12.4.0/include/c++/12/string:53:
/usr/local/Cellar/gcc@12/12.4.0/include/c++/12/cstdio:98:11: error: 'FILE' has not been declared in '::'
   98 |   using ::FILE;
      |           ^~~~
~~~
Any vcpkg file involved at all? Maybe a bug in homebrew's gcc. 
Not vcpkg's supported x64-osx (Apple clang).

**Comment by mbartu-r7** [comment]
The issue is related to MacOSX SDK 15 - it won't replicate for SDK 13.3 . It is not related just to Vcpkg, @MonicaLiu0311 I do apologize for wrong addressing.

I'm not sure it happens for clang as well, I tried only on gcc@12.

My Arm-based mac was not updated for 15, and since by default cmake takes the last one, I was able to see that issue only on Intel-based mac that was updated.
